### PR TITLE
Added internal api tests

### PR DESCRIPTION
**Pull request by witlessbird**
This is to catch realm module internal api changes

### CODE REVIEW

File: test/internal_api_test.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+require 'test_helper'
+require "rack/test"
```
Comment (timogoebel) [inline test/internal_api_test.rb]: Nit: Singlequotes?

File: Gemfile [hunk 2]
```diff
@@ -2,5 +2,11 @@ source 'https://rubygems.org'
...
 
 group :development do
-  gem 'smart_proxy', :git => 'https://github.com/theforeman/smart-proxy.git', :branch => 'develop'
+  if RUBY_VERSION < '2.2.2'
+    gem 'rack-test', '~> 0.7.0'
+  else
+    gem 'rack-test'
+  end
+
+  gem 'smart_proxy', :github => 'theforeman/smart-proxy', :branch => 'develop'
```
Comment (timogoebel) [inline Gemfile]: This does not work behind a proxy. Mind restoring the original version?

**Comment by witlessbird** [comment]
* reverted smart-proxy dependency requirement to use :git
* switched to single quotes for ```rack/test``` require

**Comment by witlessbird** [comment]
I think it's acceptable here to merge ones own commits once they have been approved?

**Pull request merged**